tests/test_mab.py

```diff
@@ -1211,3 +1252,300 @@ def test_convert_matrix(self):
         MAB._convert_matrix(c.loc[0], row=True)
         MAB._convert_matrix(d)
 
+    #################################################
+    # Test serialization
+    ################################################
+
+    def test_pickle_before_fit(self):
+        rng = np.random.RandomState(seed=9)
+        train_data = pd.DataFrame({'a': [rng.rand() for _ in range(20)],
+                                   'b': [rng.rand() for _ in range(20)],
+                                   'c': [rng.rand() for _ in range(20)],
+                                   'decision': [1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2],
+                                   'reward': [0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 1, 1]})
+        test_data = pd.DataFrame({'a': [rng.rand() for _ in range(3)], 'b': [rng.rand() for _ in range(3)],
+                                  'c': [rng.rand() for _ in range(3)], 'decision': [1, 1, 2], 'reward': [0, 1, 1]})
+        context_columns = ['a', 'b', 'c']
+
+        for lp in BaseTest.lps:
+            mab = MAB([1, 2], lp)
+            file = open('mab.pkl', 'wb')
+            pickle.dump(mab, file)
+            file.close()
+
+            file2 = open('mab.pkl', 'rb')
+            new_mab = pickle.load(file2)
+            file2.close()
+
+            new_mab.fit(train_data['decision'], train_data['reward'])
+            new_mab.predict()
```

Comment:
Can you also add a test where the the output of predict() before pickling and the output of predict() after loading are identical?